Tools/PowershellModule/src/Microsoft.WinGet.Source.psm1

```diff
@@ -1,50 +1,57 @@
+#
+# Module script for module 'Microsoft.WinGet.Source'
+#
 
 ## Loads the binaries from the Desktop App Installer Library - Only if running PowerShell at a specified edition
-if ($PSEdition -eq $WinGetPSEdition) {
-        try {
-                Add-Type -Path "$PSScriptRoot\Library\WinGet.RestSource.PowershellSupport\Microsoft.Winget.PowershellSupport.dll"
-                $WinGetDesktopAppInstallerLibLoaded=$true 
+try {
+    $architecture = [System.Runtime.InteropServices.RuntimeInformation]::ProcessArchitecture
+    
+    switch ($architecture) {
+        "X64" {
+            Copy-Item -Path "$PSScriptRoot\Library\WinGet.RestSource.PowershellSupport\runtimes\win-x64\native\WinGetUtil.dll" -Destination "$PSScriptRoot\Library\WinGet.RestSource.PowershellSupport\WinGetUtil.dll" -Force
+        }
+        "X86" {
+            Copy-Item -Path "$PSScriptRoot\Library\WinGet.RestSource.PowershellSupport\runtimes\win-x86\native\WinGetUtil.dll" -Destination "$PSScriptRoot\Library\WinGet.RestSource.PowershellSupport\WinGetUtil.dll" -Force
+        }
+        "Arm64" {
+            Copy-Item -Path "$PSScriptRoot\Library\WinGet.RestSource.PowershellSupport\runtimes\win-arm64\native\WinGetUtil.dll" -Destination "$PSScriptRoot\Library\WinGet.RestSource.PowershellSupport\WinGetUtil.dll" -Force
         }
-        catch [System.Reflection.ReflectionTypeLoadException] {
-                Write-Host "Message: $($_.Exception.Message)"
-                Write-Host "StackTrace: $($_.Exception.StackTrace)"
-                Write-Host "LoaderExceptions: $($_.Exception.LoaderExceptions)"
-                throw $_
+        Default {
+            throw "Powershell Core runtime architecture not supported"
         }
+    }
+    
+    Add-Type -Path "$PSScriptRoot\Library\WinGet.RestSource.PowershellSupport\Microsoft.Winget.PowershellSupport.dll"
 }
-else {
-        throw "Unable to load required binaries. Verify you are using Powershell $($WinGetPSEdition)."
+catch {
+    ## Exceptions thrown by Add-Type will not fail the Import-Module. Catch and re-throw to fail the Import-Module.
+    throw $_
 }
 
+## Load classes first
+. $PSScriptRoot\Library\WinGetManifest.ps1
+
+## Loads Libraries
+Get-ChildItem -Path "$PSScriptRoot\Library" -Filter *.ps1 | foreach-object { . $_.FullName }
+
 ## Validates that the required Azure Modules are present when the script is imported.
-[string[]]$RequiredModules = @("Az.Resources", "Az.Accounts", "Az.Websites", "Az.Functions", "powershell-yaml")
+[string[]]$RequiredModules = @("Az")
 
-## Verifies that the Azure Modules were previously installed.
-[Boolean] $TestResult = Test-PowerShellModuleExist -Modules $RequiredModules
-if (!$TestResult) { 
-        ## Installs the required Azure Modules if not already installed
-        $RequiredModules | ForEach-Object { Install-Module $_ -Force }
+foreach ($RequiredModule in $RequiredModules) {
+    ## Tests if the module is installed
+    $Result = Test-PowerShellModuleExist -Name $RequiredModule
```

Comment:
nit: I would add a log here that prints the required module so the user knows what is needed in case this fails.

Reply:
They are logged in the Test-PowerShellModuleExist function if not found